frontend/src/pages/modelCatalog/screens/RegisterCatalogModel.tsx

```diff
@@ -109,24 +106,19 @@ const RegisterCatalogModel: React.FC = () => {
       setData('modelLocationURI', model.artifacts?.map((artifact) => artifact.uri)[0] || '');
       setData('modelRegistry', preferredModelRegistry?.metadata.name || '');
 
-      const registeredFromReferenceCustomProperties: ModelRegistryCustomProperties = Object.entries(
-        decodedParams,
-      )
-        .filter(([key]) => /^\w+$/.test(key))
-        .reduce((acc: ModelRegistryCustomProperties, [key, value]) => {
-          if (typeof value === 'string') {
-            acc[`_registeredFromCatalog${capitalize(key)}`] = {
-              // eslint-disable-next-line camelcase
-              string_value: value,
-              metadataType: ModelRegistryMetadataType.STRING,
-            };
-          }
-          return acc;
-        }, {});
+      // Use the utility function to convert catalog params to model source properties
+      const sourceProperties = catalogParamsToModelSourceProperties(decodedParams);
+      // Set additional artifact properties in one object
+      setData('additionalArtifactProperties', {
+        ...sourceProperties,
+        modelSourceName: model.name,
+        modelSourceId:
+          model.artifacts?.map((artifact) => artifact.tags && artifact.tags[0])[0] ?? '',
+      });
```

Comment:
I don't think you need to define modelSourceName and modelSourceId again here right? The values you get here will be the same as the ones in `decodedParams` which are already included in `sourceProperties`.